sled-agent/src/services.rs

```diff
@@ -456,6 +405,21 @@ impl ServiceManager {
                         }
                     }
                 }
+                ServiceType::Tfport { pkt_source } => {
+                    // The tfport service requires a MAC device to/from which sidecar
+                    // packets may be multiplexed.  If the link isn't present, don't
+                    // bother trying to start the zone.
```

Comment:
I'm not really able to follow all the logic here, but will this be retried? The `tfpkt` driver can be reloaded, or otherwise manually manipulated, especially during development. If there is no `pkt_source` at one point in time, this will be retried, right?

Reply:
Here's the callstack, starting from the `pub` API used elsewhere:

`pub activate_switch` <- Called when we detect the tofino driver was loaded
`ensure_switch_zone`
`start_switch_zone`
`initialize_zone_loop` <- Called in a newly spawned `tokio::task`, loops until the zone launches or we're explicitly told to stop.
`initialize_zone`
`link_needed` <- This is the function where we're verifying the `pkg_source`

So yeah, we'll keep retrying to call `initialize_zone` forever until we're told otherwise (e.g., if the tofino driver is unloaded).